### ISSUE TITLE
One-hot encoded categorical data and sklearns RF, XGBoost don't work properly.

### ISSUE THREAD

**Issue by DnanaDev** [DnanaDev/Delhi_Rental_Price_Prediction#2]
There seems to be different opinions about using one-hot encoded categorical features with implementations that don't natively support them. Try **CatBoost** or H20 Random Forrest that support categorical data by design. Also, investigate one-hot encoding not being recommended for features with high cardinality, something to do with creating very sparse features. 

For reference : 
https://roamanalytics.com/2016/10/28/are-categorical-variables-getting-lost-in-your-random-forests/
https://medium.com/data-design/visiting-categorical-features-and-encoding-in-decision-trees-53400fa65931
https://www.kaggle.com/c/avito-demand-prediction/discussion/57094
https://www.kaggle.com/c/zillow-prize-1/discussion/38793
https://www.kaggle.com/c/bnp-paribas-cardif-claims-management/discussion/19851

Resolution : 
Looking at the Feature Importance for the models. 
1. Random Forrest - just seems to put all importance on the size feature and the categorical features are drowned out.
<img width="771" alt="Screenshot 2020-07-18 at 2 20 37 PM" src="https://user-images.githubusercontent.com/47149256/87849149-e575aa00-c903-11ea-82c9-5f512645b48c.png">
2. XGBoost Regressor - Seems to put more importance on the Categorical features like South Delhi, this is good, but worse validation set performance than  RF. 
<img width="809" alt="Screenshot 2020-07-18 at 2 19 57 PM" src="https://user-images.githubusercontent.com/47149256/87849161-f0303f00-c903-11ea-8f15-415a2a5dc57e.png">
3. CatBoost - Best performing individual model. Can't directly compare feature importance to other models, but puts significant importance on the categorical features. Good option.
<img width="730" alt="Screenshot 2020-07-18 at 2 20 13 PM" src="https://user-images.githubusercontent.com/47149256/87849164-f9b9a700-c903-11ea-9c74-9a3c0c29c3a2.png">
